phoenix-core-server/src/main/java/org/apache/phoenix/replication/log/LogFile.java

```diff
@@ -0,0 +1,439 @@
+         * @param tableName The HBase table name to set.
+         * @return This Record instance for chaining.
+         */
+        Record setHBaseTableName(String tableName);
+
+        /**
+         * Gets the commit ID or System Change Number (SCN) associated with this record.
+         * @return The commit ID/SCN.
+         */
+        long getCommitId();
+
+        /**
+         * Sets the commit ID or System Change Number (SCN) for this record.
+         * @param commitId The commit ID/SCN to set.
+         * @return This Record instance for chaining.
+         */
+        Record setCommitId(long commitId);
+
+        /**
+         * Gets the total serialized length of this record in bytes, including any length prefixes
+         * used by the codec. This value should be set by the codec after writing or reading.
+         * @return The total serialized length of the record.
+         */
+        int getSerializedLength();
+
+        /**
+         * Sets the total serialized length of this record. This is typically called by the codec.
+         * @param serializedLength The total serialized length in bytes.
+         * @return This Record instance for chaining.
+         */
+        Record setSerializedLength(int serializedLength);
+    }
+
+    /** Interface for writing replication logs */
+    interface Writer extends Closeable {
+        /**
+         * Initializes the writer with necessary context.
+         * @param context The context containing initialization parameters.
+         * @throws IOException if initialization fails.
+         */
+        void init(LogFileWriterContext context) throws IOException;
+
+        /**
+         * Appends an HBase mutation to the log file. The log record may be buffered internally.
+         * @param tableName The HBase table name
+         * @param commitId The commit identifier
+         * @param mutation The mutation to append.
+         * @throws IOException if an I/O error occurs during append.
+         */
+        void append(String tableName, long commitId, Mutation mutation) throws IOException;
+
+        /**
+         * Flushes any buffered data to the underlying storage and ensures it is durable
+         * (e.g., by calling hsync on the FSDataOutputStream). This guarantees that records
+         * appended before the sync are persisted.
+         * @throws IOException if an I/O error occurs during sync.
+         */
+        void sync() throws IOException;
+
+        /**
+         * Returns the current length of the log file being written, in bytes.
+         * This typically reflects the position of the underlying output stream.
+         * @return The current file length.
+         * @throws IOException if the length cannot be determined.
+         */
+        long getLength() throws IOException;
+    }
+
+    /** Interface for reading replication logs */
+    interface Reader extends Closeable, Iterable<Mutation> {
```

Comment:
It is not sufficient just to read the mutation part of a record. Phoenix needs to know the HBase table name for this mutation.

Reply:
That's why originally I had this returning Record :-)

Reply:
Let me put that back...